### PR TITLE
Fix: Adding semantic-release package and requirements to the project [#167]

### PR DESCRIPTION
**Pull request by cahirodoherty-learningpool**
### Issue ID

Addresses #167 

### Context/Purpose

Adding semantic-release package and requirements to the project. This allows pack version bumping, tagging and release notes to be automated upon code merge (via PR)

### Implementation details

Followed the steps outlined here: https://github.com/semantic-release/semantic-release#readme

### PR Checklist

- [ ] The title begins with the semantic-release format "type(Name): "
- [ ] This is followed by a description of the task
- [ ] I have added the issue number(s) to the title
- [ ] The title matches the following format "type(Name): Description of the task [#123]"

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+### Implementation details
+
+An overview of the approach taken to action the issue
+
+### PR Checklist
+
+- [ ] The title begins with the semantic-release format "type(Name): "
+- [ ] This is followed by a description of the task
+- [ ] I have added the issue number(s) to the title
+- [ ] The title matches the following format "type(Name): Description of the task [#123]"
```
Comment (oliverfoster) [inline .github/pull_request_template.md:18]:
```suggestion
fixes #123

### Fixed
* A sentence describing each fix

### Added
* A sentence describing each addition

### Testing
1. Steps for testing

requires #123 

### PR Checklist
- [ ] The title begins with the semantic-release format "type(Name): "
- [ ] This is followed by a description of the task
- [ ] I have added the issue number(s) to the title
- [ ] The title matches the following format "type(Name): Description of the task [#123]"
```

**Comment by oliverfoster** [comment]
We're going to need to look at bower.json:version auto increment as well as pacakge.json for the other plugins

File: .github/pull_request_template.md (hunk 1 above)
Comment (oliverfoster) [reply .github/pull_request_template.md:18]: Not sure about the checklist part. I understand that their needs to be title enforcement for automation, but a fillable checklist seems a little too much. Any way to prepopulate it? Or at least shorten that text to something more deletable/informative.

Comment (cahirodoherty-learningpool) [reply .github/pull_request_template.md]: I've made some changes and put a lot of the descriptive elements in comments, so it should reduce the output noise. Can you take a look please?

**Comment by oliverfoster** [comment]
So. How/when does this actually release? Is it as soon as a pr gets merged into master, it'll bump and publish?

File: package.json [hunk 2]
```diff
@@ -12,5 +12,33 @@
...
+  "private": true,
+  "devDependencies": {
+    "@semantic-release/git": "^10.0.1",
+    "conventional-changelog-eslint": "^3.0.9",
+    "semantic-release": "^19.0.3"
+  },
+  "release": {
+    "plugins": [
+      "@semantic-release/release-notes-generator",
+      "@semantic-release/npm",
```
Comment (oliverfoster) [inline package.json:25]: We don't publish these to npm yet.

File: package.json [hunk 3]
```diff
@@ -12,5 +12,33 @@
...
+      [
+        "@semantic-release/commit-analyzer",
+        {
+          "preset": "eslint"
+        }
+      ],
+      [
+        "@semantic-release/git",
+        {
+          "assets": [
```
Comment (oliverfoster) [inline package.json:36]: I'm not sure you need the assets section. That would attach the package.json file to the release as an attachment, i believe?

File: package.json (hunk 2 above)
Comment (oliverfoster) [reply package.json:25]: https://semantic-release.gitbook.io/semantic-release/extending/plugins-list

File: package.json (hunk 3 above)
Comment (oliverfoster) [reply package.json:36]: You may not even need the git plugin? https://github.com/semantic-release/git

Comment (oliverfoster) [reply package.json:36]: https://semantic-release.gitbook.io/semantic-release/support/faq#why-is-the-package.jsons-version-not-updated-in-my-repository This comment is conflicting

**Comment by oliverfoster** [comment]
[  "scripts": {
    "postversion": "cp -r package.json .. && cp -r npm-shrinkwrap.json .."
  }](https://github.com/semantic-release/npm#examples)
  
  The npm readme says we can probably use "scripts" to sync the bower.json with the package.json
  
 `"postversion": "cp package.json bower.json"` should do it?

File: package.json (hunk 2 above)
Comment (cahirodoherty-learningpool) [reply package.json:25]: Hi Oli, documentation had told me that the `"private": true` package.json setting bypasses the npm publish. But I did think it still had to be included. I'll test if it works without that package

File: package.json (hunk 3 above)
Comment (cahirodoherty-learningpool) [reply package.json:36]: Yeah my understanding here was that for commit abilities, the @semantic-release/git package needed to be used, and 'release' files/folders needed specified in the assets array. i.e files that will be updated post-merge, pre-tag

**Comment by cahirodoherty-learningpool** [comment]
Hi Oli, just to address your comments collectively;

I have confirmed that the npm and git plugins are both needed for package.json updates to be included in the tag.
The `"private": true` setting on package.json avoids any attempt to actually post an npm package.
The `assets` array targets permitted file changes post-merge, pre-release.
The addition of
```"scripts": {
  "postversion": "cp package.json bower.json"
}```
does indeed work for plugins. The only further addition required is that `bower.json` be added to the assets array

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**